track/trial.py

```diff
@@ -74,16 +127,33 @@ def metric(self, *, iteration=None, **kwargs):
         for hook in self._hooks:
             hook.on_result(new_args)
 
-    def artifact(self, artifact_name, src):
-        srcpath = os.path.expanduser(src)
-        # Copy filepath to the base_dir
-        destpath = os.path.join(self.artifact_dir, artifact_name)
-        os.path.makedirs(os.path.dirname(artifact_name), exist_ok=True)
-        shutil.copy(srcpath, destpath)
+    def artifact_directory(self):
+        """returns the local file path to the trial's artifact directory"""
+        return self.artifact_dir
 
     def close(self):
         for hook in self._hooks:
             hook.close()
+        # unsure if editting the param_map file like this is very kosher
+        self.param_map["trial_completed"] = True
+        # using a constructor for its side effect here (updating the param map)
+        UnifiedLogger(
+            self.param_map, self.data_dir, self.trial_id + "_").close()
 
     def get_result_filename(self):
         return os.path.join(self.data_dir, self.trial_id + "_" + RESULT_SUFFIX)
+
+def _git_hash():
```

Comment:
can you move this into utilities file like `autodetect`?

Reply:
done